### ISSUE TITLE
Memory Leak with Savestate Loading

### ISSUE THREAD

**Issue by ADFairbanks** [citra-emu/citra#5523]
TL;DR: The binary_iarchive doesn't seem to be fully emptying its buffer on load. I've compiled Citra myself and tried a few things but couldn't get it to fully empty.

Here's my citra about for versioning:
![citra](https://user-images.githubusercontent.com/21269572/91293722-af72e380-e755-11ea-9970-3d2e17a886ed.PNG)
I checked if there were any changes in savestates with the recent canary build, and there wasn't so I didn't bother with that.

Seeing that there was savestate functionality newly added, I was excited to be able to do RNG manipulation for SMT: Strange Journey Redux speedrun routing. Unfortunately, state loading will move the memory address of the RNG values I had found. When I tried to search for it again with a constant value next to the RNG, I came up with increasingly more signatures for each load.
No savestate loads:
![no load](https://user-images.githubusercontent.com/21269572/91293297-07f5b100-e755-11ea-85cf-a706e7dd78fa.PNG)
One savestate load:
![one load](https://user-images.githubusercontent.com/21269572/91293319-0deb9200-e755-11ea-8a18-d9d29d67f7f7.PNG)
Five savestate loads:
![five load](https://user-images.githubusercontent.com/21269572/91293346-16dc6380-e755-11ea-9b37-fd08dc1cfb52.PNG)
After enough loads, Citra starts to slow down and eventually crashes.
Interestingly enough, that signature is the exact point where the leak starts. If it matters, that location starts at 0x003a513c in 3DS memory (verified on console hardware).

I then built and compiled Citra myself to see if I could solve the issue myself. I took memory snapshots before and after a savestate load in Visual Studio:
![memory diff](https://user-images.githubusercontent.com/21269572/91294443-bd753400-e756-11ea-8917-bd0c3fff2cd9.PNG)
and inspecting that extra 50MB I found the structure that's using it inside of the System::LoadState() function:
![memory diff 2](https://user-images.githubusercontent.com/21269572/91294983-979c5f00-e757-11ea-97d9-b7847f65bfb5.PNG)
Unfortunately I have no experience with C++ Boost libraries so I have no idea how I could empty the buffer of the stream / archive, even if that is the problem. I tried emptying the stringstream with sstream.str("");, but that did nothing. Everything else I tried either did nothing or crashed.
What might have to happen is to have serialization done using a different library, but who knows if that would even fix it.
